springfield/firefox/tests/test_views.py

```diff
@@ -366,25 +366,56 @@ def test_post(self, render_mock):
+            ("en-US", False, True, "legacy", "firefox/download/desktop/download.html"),  # legacy gets desktop template
+            # and again, for non-English
+            ("de", True, True, None, "firefox/download/home.html"),  # default
+            ("de", True, True, "basic", "firefox/download/basic/base_download.html"),  # basic experience
+            ("de", True, True, "legacy", "firefox/download/desktop/download.html"),  # legacy experience
+            ("de", True, True, "watermelon", "firefox/download/home.html"),  # experience doesn't exist
+            ("de", False, False, None, "firefox/download/basic/base_download.html"),  # No FTL is active
+            ("de", False, True, None, "firefox/download/desktop/download.html"),  # desktop FTL active, home FTL inactive
+            ("de", False, True, "basic", "firefox/download/basic/base_download.html"),  # basic always gets basic template
+            ("de", False, True, "legacy", "firefox/download/desktop/download.html"),  # legacy gets desktop template
+        ]
+
+        for case in test_cases:
+            locale, home_ftl_active, desktop_ftl_active, experience, expected_template = case
             with self.subTest(
                 locale=locale,
-                ftl_file_is_active_value=ftl_file_is_active_value,
+                home_ftl_active=home_ftl_active,
+                desktop_ftl_active=desktop_ftl_active,
+                experience=experience,
                 expected_template=expected_template,
             ):
-                with patch.object(views, "ftl_file_is_active", lambda *x: ftl_file_is_active_value):
-                    req = RequestFactory().get("/")
+                # Create a mock that returns different values based on the FTL file
+                def ftl_active_mock(ftl_file):
+                    if ftl_file == "firefox/download/home":
+                        return home_ftl_active
+                    elif ftl_file == "firefox/download/desktop":
+                        return desktop_ftl_active
+                    return False
+
+                with patch.object(views, "ftl_file_is_active", ftl_active_mock):
+                    req = RequestFactory().get("/", {"xv": experience} if experience else {})
                     req.locale = locale
                     view = views.DownloadView.as_view()
                     view(req)
                     template = render_mock.call_args[0][1]
-                    assert template == [expected_template]
+                    assert template == [expected_template], (
+                        f"Template mismatch for locale={locale}, home_ftl_active={home_ftl_active}, "
+                        f"desktop_ftl_active={desktop_ftl_active}, experience={experience}.\n"
+                        f"Expected: [{expected_template}]\n"
+                        f"Actual: {template}"
+                    )
```

Comment:
Smart!

I did test a few of these under different conditions and don't see any flaw in that logic, so while it may need an extra pythonista check 👓 I'd say this is safe to land.